### PR TITLE
Refactoring, generate ESM

### PR DESCRIPTION
**Pull request by jscheid**
- No functional changes
- Generate ESM modules
- Upgrade dependencies
- Remove runtime dependency on `typed-assert`
- Restructure for cleaner main directory
- Improve code readability
- Add more tests

### CODE REVIEW

**Comment by jscheid** [comment]
Hi @MLoughry, I'm going to add you as a reviewer here because I've made some fairly extensive changes to your feature: 9a9ffab1d5124784e78976f6178d4dc57ee84f8e and 20a2d7a35610d71416f3f2a9936523dadeb603ce. This was mainly to make CodeClimate happy, and even if I don't agree with all of its mandates (for example, I thought your use of nested functions was perfectly fine) I do think it's a slight improvement overall. Let me know what you think.

Thanks to the typings and test coverage I'm fairly confident that I didn't break anything but you might want to check by running it against your product.

Feel free to provide feedback on the rest of the changes as well, but don't feel obliged. Thanks!

File: wsi-test-helper/package.json [hunk 1]
```diff
@@ -2,24 +2,23 @@
   "name": "wsi-test-helper",
   "version": "1.0.0",
   "description": "Helpers for webpack-subresource-integrity tests",
-  "main": "index.js",
+  "main": "dist/index.js",
   "license": "MIT",
   "private": true,
   "devDependencies": {
-    "@tsconfig/node10": "^1.0.8",
+    "@tsconfig/node12": "^1.0.9",
```
Comment (trumbitta) [inline wsi-test-helper/package.json:9]: We should update Node in this project too, sooner or later, I guess 😺

**Review by trumbitta — APPROVED** [review]
LGTM :shipit: 

Only serious comment I have is the first one, about `tsx` 😃

File: nyc-apply-source-map/index.js [hunk 2]
```diff
@@ -30,7 +30,17 @@ const main = async () => {
   const files = await globPromise(join(inDir, "*.json"));
   await Promise.all(
     files.map(async (file) => {
-      inputCoverageMap.merge(JSON.parse(await readFilePromise(file, "utf-8")));
+      const data = JSON.parse(await readFilePromise(file, "utf-8"));
+      const isTypeScript = Object.keys(data)[0]?.endsWith(".ts");
```
[the remark this answers is not in the document]
Comment (jscheid) [reply nyc-apply-source-map/index.js:34]: It doesn't need to, no, but thanks for the heads up.

File: wsi-test-helper/package.json (hunk 1 above)
Comment (jscheid) [reply wsi-test-helper/package.json:9]: It's the minimum version and no need to update it at this point.

**Pull request merged**